File: experiment/advent-of-code-2022/day1/main.py
#!/usr/bin/env python3

# python3 main.js
import sys
from day1 import *

# sys.setrecursionlimit(21000)  # required to prevent stackoverflow b/c python doesn't do tail recursion

file = open('food.txt', 'rb')
food = file.read()
file.close()

res = day1a(list(food))
print(res)

File: experiment/advent-of-code-2022/day1/main.lua
#!/usr/bin/env lua

-- lua main.lua
require "day1"

local function read_file(path)
    local file = io.open(path, "rb") -- r read mode and b binary mode
    if not file then return nil end
    local content = file:read "*a" -- *a or *all reads the whole file
    file:close()
    return content
end

local food = read_file("food.txt")

function stringToList(str)
    if str == "" then
        return {}
    else
        return { head = string.byte(string.sub(str, 1, 1)), tail = stringToList(string.sub(str, 2)) }
    end
end

foodList = stringToList(food)
res = day1a(foodList)
print(res)
